game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java

```diff
@@ -228,16 +223,13 @@ private void joinGame() {
     // we sort the table, so get the correct index
     final GameDescription description =
         gameTableModel.get(gameTable.convertRowIndexToModel(selectedIndex));
-    GameRunner.joinGame(description, messengers);
+    GameRunner.joinGame(description, lobbyClient.getPlayerName());
```

Comment:
Off-Topic:
I have been thinking about what needs to be done in order to move more classes up to `game-headed`, and one of the first things that need to be done is getting rid of all the static helper methods in `GameRunner` because this class is the root of all the headed code.

Besides that, spawning a new JVM to launch a game is still a bad idea IMO, but unfortunately the whole game logic relies on it currently. (Too many `System.exit(0)`s everywhere)

Reply:
We also *need* to get rid of the JVM launch to be listed on Steam. I've been wondering if we should go ahead and do the game listing even if we don't get the UI refresh complete and work on that post-Steam launch. Then again, the JVM launch is a hard blocker for it.

TripleAFrame is a similar one to GameRunner, between there is some ground to cover but would go a long ways.

The code that has been moved into game-headed and lobby has, I've found, been really helpful.

I've debated a bit personally whether we want a 3rd package, one for the UI layer. Essentially we'd have "game-headed-swing" > "game-logic" > "game-core". The top layer would inject interface implementations that bind the UI to swing. I suspect it's maybe premature to go that far, but it's something I've been considering in the back of my mind.